### PR TITLE
add fieldset and legend to FormStep component

### PR DESCRIPTION
**Pull request by senadir**
[Fieldset and Legend](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) help group a set of inputs in a semantic way, this PR changes some divs to be `<fieldset>` and `<legend>`

### CODE REVIEW

**Comment by senadir** [comment]
@joshuatf are you suggesting we remove the `<legend>` element replace it with an area thing? also, wouldn't `screen-reader-text` hide the text?

**Comment by joshuatf** [comment]
>are you suggesting we remove the <legend> element replace it with an area thing?

@senadir Not quite.  My suggestion was to potentially add an additional prop for `legend` instead of using the description or title if neither of these fit well.

>also, wouldn't screen-reader-text hide the text?

It would.  The idea here would be to provide screen reader text via the legend (if different from the title) and keep the title and description separate for sighted users.

I think that using the title in place of description for the legend means we don't have to do this, but if you find that the titles don't work well when read before each input, as a screen reader would read them, then we might consider a separate prop for the legend.

**Comment by senadir** [comment]
> Is it okay to assume that every step in the checkout will require a fieldset?

Yes, every step in the checkout **form** will have some kind of fields to it, inputs, radio, checkbox, any extra information that is not meant to submit or collect the info should be in the sidebar.

> Is it okay to assume that every stepHeading will have a legend?

I used `title` when no legend text is present, basically, each step should have a title describing it, and we would define generic steps for people to hook into + the ability to introduce new steps, once we do that, we can make title required

File: assets/js/base/components/checkout/form-step/index.js [hunk 1]
```diff
@@ -29,9 +29,12 @@ const StepNumber = ( { stepNumber } ) => {
 	);
 };
 
-const StepHeading = ( { title, stepHeadingContent } ) => (
+const StepHeading = ( { title, legend, stepHeadingContent } ) => (
 	<div className="wc-components-checkout-step__heading">
-		<h4 className="wc-components-checkout-step__title">{ title }</h4>
+		<h4 className="wc-components-checkout-step__title">
+			<legend className="screen-reader-text">{ legend || title }</legend>
```
Comment (joshuatf) [inline assets/js/base/components/checkout/form-step/index.js]: Nice!  Like the legend option with the title fallback.

I think this legend cannot be inside an `h4` and needs to come immediately after the opening `<fieldset>` tag to be semantically correct.  Checked using Voiceover and the legend currently isn't read.

**Comment by nerrad** [comment]
In the latest commits,

- `legend` is moved to be right after the fieldset in keeping with standards.
- I removed a background that was on fieldset (following design)
- I removed a left border and content width on the form that was causing overflow on smaller screen-sizes and makes the form more responsive. This also preps the form for when we add the order summary. It's important to have flexible widths instead of static.

I think this is ready for a final review and we can get it merged.

File: assets/js/base/components/checkout/form-step/index.js [hunk 2]
```diff
@@ -31,7 +31,9 @@ const StepNumber = ( { stepNumber } ) => {
 
 const StepHeading = ( { title, stepHeadingContent } ) => (
 	<div className="wc-block-checkout-step__heading">
-		<h4 className="wc-block-checkout-step__title">{ title }</h4>
+		<h4 className="wc-block-checkout-step__title">
+			<span aria-hidden="true">{ title }</span>
```
Comment (Aljullu) [inline assets/js/base/components/checkout/form-step/index.js]: Could the `aria-hidden` be moved to the `<h4>` tag, instead?

Comment (nerrad) [reply assets/js/base/components/checkout/form-step/index.js]: ya and with that the span is unnecessary.

**Pull request merged**